Pod/Classes/NJHNavigationBarBackgroundSelectionView.m

```diff
@@ -0,0 +1,173 @@
+//
+//  NJHNavigationBarBackgroundSelectionView.m
+//  Pods
+//
+//  Created by Hakon Hanesand on 7/29/15.
+//
+//
+
+#import "NJHNavigationBarBackgroundSelectionView.h"
+
+/**
+ *  This is the path to the framework bundle generated by Cocoapods that our xib is contained in. See the instance method for usage
+ */
+static NSString * const kNJHFrameworkBundleName = @"/Frameworks/NJHNavigationBarSelectorPageViewController.framework/";
+
+/**
+ *  The space that is inserted between the selection view and the background view on all sides
+ */
+static int const kNJHSelectionViewSpacerSize = 1;
+
+/**
+ *  Default font size for labels
+ */
+static int const kNJHDefaultFontSize = 10;
+
+@interface NJHNavigationBarBackgroundSelectionView ()
+
+/**
+ *  The constraint that controls the right to left placement of the selector view. This constraint is anchored to the left side of this view and to
+ *  the center x of the selectorView. It is the constraint we manipulate to get the view to move right to left
+ */
+@property (weak, nonatomic) IBOutlet NSLayoutConstraint *animateableConstraint;
+
+/**
+ *  The number of sections this selection view should display, calculated from the labelTitles array
+ */
+@property (nonatomic) NSUInteger sections;
+
+/**
+ *  One for each section on the
+ */
+@property (nonatomic) NSMutableArray *labels;
+@end
+
+@implementation NJHNavigationBarBackgroundSelectionView
+
+/**
+ *  Since we are overriding both the setter and the getter for the following properties, we must synthesize them
+ */
```

Comment:
You are right as discussed, there are some exceptions 
http://stackoverflow.com/questions/19784454/when-should-i-use-synthesize-explicitly